qiskit/compiler/transpiler.py

```diff
@@ -561,128 +319,144 @@ def _parse_transpile_args(
-        "init_method": init_method,
-        "optimization_method": optimization_method,
-    }
-
-    list_transpile_args = []
+
     if not ignore_backend_supplied_default_methods:
         if scheduling_method is None and hasattr(backend, "get_scheduling_stage_plugin"):
             scheduling_method = backend.get_scheduling_stage_plugin()
         if translation_method is None and hasattr(backend, "get_translation_stage_plugin"):
             translation_method = backend.get_translation_stage_plugin()
 
-    for key, value in {
-        "inst_map": inst_map,
-        "coupling_map": coupling_map,
-        "backend_properties": backend_properties,
-        "approximation_degree": approximation_degree,
-        "initial_layout": initial_layout,
-        "layout_method": layout_method,
-        "routing_method": routing_method,
-        "translation_method": translation_method,
-        "scheduling_method": scheduling_method,
-        "instruction_durations": durations,
-        "timing_constraints": timing_constraints,
-        "seed_transpiler": seed_transpiler,
-        "unitary_synthesis_method": unitary_synthesis_method,
-        "unitary_synthesis_plugin_config": unitary_synthesis_plugin_config,
-        "target": target,
-        "hls_config": hls_config,
-    }.items():
-        if isinstance(value, list):
-            # This giant if-statement detects deprecated use of argument
-            # broadcasting. For arguments that previously supported broadcast
-            # but were not themselves of type list (the majority), we simply warn
-            # when the user provides a list. For the others, special handling is
-            # required to disambiguate an expected value of type list from
-            # an attempt to provide multiple values for broadcast. This path is
-            # super buggy in general (outside of the warning) and since we're
-            # deprecating this it's better to just remove it than try to clean it up.
-            # pylint: disable=too-many-boolean-expressions
-            if (
-                key not in {"instruction_durations", "timing_constraints", "initial_layout"}
-                or (
-                    key == "initial_layout"
-                    and user_input_initial_layout
-                    and isinstance(user_input_initial_layout, list)
-                    and isinstance(user_input_initial_layout[0], (Layout, dict, list))
-                )
-                or (
-                    key == "instruction_durations"
-                    and user_input_durations
-                    and isinstance(user_input_durations, list)
-                    and isinstance(user_input_durations[0], (list, InstructionDurations))
+    if instruction_durations or dt:
+        # If durations are provided and there is more than one circuit
+        # we need to serialize the execution because the full durations
+        # is dependent on the circuit calibrations which are per circuit
```

Comment:
To me, this seems to be indicating a problem with the passes that interpret this data; they should read it from the circuit during their `run` methods.  It's the same problem that we had with `SetLayout` until just recently.

Ideally we'd fix that before this PR, but there's no time before 0.25 now.  Can we check with the people who might be using these options within IBM that they're not going to be hamstrung by the decay to serial transpile, at least?

Reply:
It is definitely a problem. My concern with doing the same fix as with `SetLayout` was I didn't think it was used in a single place like the input layout and that `InstructionDurations` would have to be updated in an instance variable in multiple passes. So I went with this approach instead to avoid doing this in multiple places.

@nkanazawa1989 or @itoko do you have any idea/thoughts  here regarding the potential impact of serializing the execution with a custom instruction durations or dt input? I expect you've got the best handle on the impact of setting custom instruction durations (or dt) on the argument inputs to `transpile()`.

My thinking is for most users it won't be noticeable because my feeling is most of them are on macOS anyway and we disable multiprocessing by default on macOS.

Reply:
I think the potential impact of serializing the execution with a custom instruction durations or dt input would be little (at least for noisy simulation users). I didn't know until now that instruction durations were being used for pulse gates. I've been thinking `InstructionDurations` is for custom gates without calibrations used for noisy simulation and pulse gates are separately handled [here](https://github.com/Qiskit/qiskit-terra/blame/a804df58c8e59c101fc3c13ef6a1a4cb172b9a5b/qiskit/transpiler/passes/scheduling/scheduling/base_scheduler.py#L61). @naoki, do you think the case you mentioned above is covered by the code in the link without handling of `circuits` in `_parse_instruction_durations`?